packages/data/addon/models/metadata/column.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+
+import EmberObject from '@ember/object';
+import { inject as service } from '@ember/service';
+import KegService from '../../services/keg';
+import Table from './table';
+
+export type ColumnType = 'ref' | 'formula' | 'field';
+
+export type BaseExtendedAttributes = {
+  description: string;
+};
+
+export default class Column extends EmberObject {
+  /**
+   * @property {KegService} keg
+   */
+  @service('keg')
+  keg!: KegService;
+
+  /**
+   * @property {string} id
+   */
+  id!: string;
+
+  /**
+   * @property {string} name - Display name
+   */
+  name!: string;
+
+  /**
+   * @property {string|undefined} description - an extended attribute that can be fetched
+   */
+  description?: string;
+
+  /**
+   * @property {string} tableId
+   */
+  tableId!: string;
+
+  /**
+   * @property {Table} table
+   */
+  get table(): Table {
+    const { tableId, keg, source } = this;
+    const table = keg.getById('metadata/table', tableId, source);
+    return (table as unknown) as Table;
+  }
+
+  /**
+   * @property {String} source - name of the data source this column is from.
+   */
+  source!: string;
+
+  /**
+   * @property {ColumnType} type - will be "ref", "formula", or "field" depending on where its values are sourced from
+   */
+  type!: ColumnType;
+
+  /**
+   * @property {String} expression - e.g. tableA.name if type is ref
+   */
+  expression!: string;
```

Comment:
```suggestion
   * @property {String|undefined} expression - e.g. tableA.name if type is ref
   */
  expression?: string;
```
I believe expression is only required when `type` is equal to `ref`